File: python/24.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
    def swapPairs(self, head):
        """
        :type head: ListNode
        :rtype: ListNode
        """
        dummy=ListNode(0)
        dummy.next=head
        fast=dummy
        
        while fast.next!=None and fast.next.next!=None:
            first=fast.next
            second=fast.next.next
            fast.next=second
            first.next=second.next
            second.next=first
            fast=fast.next.next
        return dummy.next

File: java/Linked_list/24.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode swapPairs(ListNode head) {
        if(head==null ||head.next==null) return head;
        ListNode cur=head, post=head.next;
        while(cur!=null&& post!=null){
            int m=cur.val;
            cur.val=post.val;
            post.val=m;
            cur=post.next;
            if (cur==null) break;
            post=cur.next;
}
        return head;
        
    }
}
#use iteration
class Solution {
    public ListNode swapPairs(ListNode head) {
        if(head==null ||head.next==null) return head;
        ListNode post_head=swapPairs(head.next.next);
        head.next.next=head;
        ListNode new_head=head.next;
        head.next=post_head;
        return new_head;
    }
}